process.c

```diff
@@ -1520,48 +1512,41 @@ rb_detach_process(rb_pid_t pid)
 
 /*
  *  call-seq:
- *     Process.detach(pid)   -> thread
+ *    Process.detach(pid) -> thread or nil
  *
- *  Some operating systems retain the status of terminated child
- *  processes until the parent collects that status (normally using
- *  some variant of <code>wait()</code>). If the parent never collects
- *  this status, the child stays around as a <em>zombie</em> process.
- *  Process::detach prevents this by setting up a separate Ruby thread
- *  whose sole job is to reap the status of the process _pid_ when it
- *  terminates. Use #detach only when you do not intend to explicitly
- *  wait for the child to terminate.
+ *  Avoids the potential for a child process to become a
+ *  {zombie process}[https://en.wikipedia.org/wiki/Zombie_process].
+ *  This method is needed only when the parent process will never wait
+ *  for the child process.
  *
- *  The waiting thread returns the exit status of the detached process
- *  when it terminates, so you can use Thread#join to
- *  know the result.  If specified _pid_ is not a valid child process
- *  ID, the thread returns +nil+ immediately.
+ *  This example does not reap the second child process;
+ *  that process appears as a zombie in the process status (+ps+) output:
  *
- *  The waiting thread has #pid method which returns the pid.
+ *    pid0 = Process.spawn('ruby', '-e', 'exit 13') # => 312691
+ *    pid1 = Process.spawn('ruby', '-e', 'exit 14') # => 312716
+ *    Process.wait(pid0)                            # => 312691
+ *    # Find zombies.
+ *    system("ps -ho pid,state -p #{pid1}")
  *
- *  In this first example, we don't reap the first child process, so
- *  it appears as a zombie in the process status display.
+ *  Output:
  *
- *     p1 = fork { sleep 0.1 }
- *     p2 = fork { sleep 0.2 }
- *     Process.waitpid(p2)
- *     sleep 2
- *     system("ps -ho pid,state -p #{p1}")
+ *     312716 Z
  *
- *  <em>produces:</em>
+ *  This example also does not reap the second child process,
+ *  but it does detach the process so that it does not become a zombie:
  *
- *     27389 Z
+ *    pid0 = Process.spawn('ruby', '-e', 'exit 13') # => 313213
+ *    pid1 = Process.spawn('ruby', '-e', 'exit 14') # => 313262
+ *    Process.wait(pid0)                            # => 313213
+ *    sleep 2
+ *    thread = Process.detach(pid1)
+ *    # => #<Process::Waiter:0x00007f038f48b838 run>
+ *    system("ps -ho pid,state -p #{pid1}")         # Finds no zombies.
  *
- *  In the next example, Process::detach is used to reap
- *  the child automatically.
+ *  The waiting thread can return the pid of the detached child process:
  *
- *     p1 = fork { sleep 0.1 }
- *     p2 = fork { sleep 0.2 }
- *     Process.detach(p1)
- *     Process.waitpid(p2)
- *     sleep 2
- *     system("ps -ho pid,state -p #{p1}")
+ *    thread.join(thread).pid                       # => 313262
```

Comment:
Did you mean to do this? `Thread#join` does not accept an argument (at least does not accept a `Thread` object as an argument).

```suggestion
 *    thread.join.pid                       # => 313262
```